core/lib/multistore/index.ts

```diff
@@ -203,21 +182,23 @@ export function localizedRouteConfig (route: RouteConfig, storeCode: string, isC
   return _route
 }
 
-export function localizedRoute (routeObj: LocalizedRoute | string | RouteConfig | RawLocation, storeCode: string = null): any {
-  if (!storeCode) {
-    storeCode = currentStoreView().storeCode
-  }
-  if (!routeObj) {
-    return routeObj
-  }
-
-  if ((typeof routeObj === 'object') && (routeObj as LocalizedRoute)) {
+export function localizedRoute (routeObj: LocalizedRoute | string | RouteConfig | RawLocation, forcedStoreCode: string = null): any {
+  // return falsy value
+  if (!routeObj) return routeObj
+  // prep route object
+  if ((typeof routeObj === 'object')) {
     if ((routeObj as LocalizedRoute).fullPath && !(routeObj as LocalizedRoute).path) { // support both path and fullPath
       routeObj['path'] = (routeObj as LocalizedRoute).fullPath
     }
   }
+  // do not make localization when multistore is off
+  if (!config.storeViews.multistore) return routeObj
```

Comment:
What about putting it together with `if (!routeObj) return routeObj`?

Reply:
Yes, I wanted to do that but previous version did that calculation even if multistore was turned off :/ 
```
  if ((typeof routeObj === 'object')) {
    if ((routeObj as LocalizedRoute).fullPath && !(routeObj as LocalizedRoute).path) { // support both path and fullPath
      routeObj['path'] = (routeObj as LocalizedRoute).fullPath
    }
  }
```
I know it doesn't sound good, but I wanted to keep same behavior as much as possible. If you want I can do that.

Reply:
Ah ok, I get it